src/server/status/cgroup.js

```diff
@@ -0,0 +1,138 @@
+import fs from 'fs';
+import { promisify } from 'bluebird';
+import { join as joinPath } from 'path';
+
+// Logic from elasticsearch/core/src/main/java/org/elasticsearch/monitor/os/OsProbe.java
+
+const CONTROL_GROUP_RE = new RegExp('\\d+:([^:]+):(/.*)');
+const CONTROLLER_SEPERATOR_RE = ',';
+
+const PROC_SELF_CGROUP_FILE = '/proc/self/cgroup';
+const PROC_CGROUP_CPU_DIR = '/sys/fs/cgroup/cpu';
+const PROC_CGROUP_CPUACCT_DIR = '/sys/fs/cgroup/cpuacct';
+
+const GROUP_CPUACCT = 'cpuacct';
+const CPUACCT_USAGE_FILE = 'cpuacct.usage';
+
+const GROUP_CPU = 'cpu';
+const CPU_FS_PERIOD_US_FILE = 'cpu.cfs_period_us';
+const CPU_FS_QUOTA_US_FILE = 'cpu.cfs_quota_us';
+const CPU_STATS_FILE = 'cpu.stat';
+
+const readFile = promisify(fs.readFile);
+
+export function readControlGroups() {
+  return readFile(PROC_SELF_CGROUP_FILE)
+    .then(data => {
+      const response = {};
+
+      data.toString().split(/\n/).forEach(line => {
+        const matches = line.match(CONTROL_GROUP_RE);
+
+        if (matches === null) {
+          return;
+        }
+
+        const controllers = matches[1].split(CONTROLLER_SEPERATOR_RE);
+        controllers.forEach(controller => {
+          response[controller] = matches[2];
+        });
+      });
+
+      return response;
+    });
+}
+
+function fileContentsToInteger(path) {
+  return readFile(path).then(data => {
+    return parseInt(data.toString());
```

Comment:
I would recommend using the radix parameter of parseInt, because without it behavior could be unpredictable. 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt